### PR TITLE
snapcraft, github: build a FIPS variant of the snap in CI

### PR DESCRIPTION
**Pull request by bboozzoo**
The branch adds support for building a FIPS variant of the snapd snap for the purpose of simply exercising that code in the github CI. The relevant patches were extracted from #13978. Note that we do not currently run any tests with that snap, nor use it for any other purpose than just having the build run and have the built snap available for download as artifact.

### CODE REVIEW

File: build-aux/snap/snapcraft.yaml [hunk 1]
```diff
@@ -206,10 +207,31 @@ parts:
...
+      # anchoring dynamic linker env variables
+      GO_TOOLCHAIN_FIPS_CHANNEL="1.18-fips/stable"
+
+      VERSION="$(craftctl get version)"
+      if [ -f fips-build ] ; then
+          # use the fips channel of Go
+          snap refresh --channel "$GO_TOOLCHAIN_FIPS_CHANNEL" go
+      fi
+      # make sure to set the version we declared in pull
+      ./mkversion.sh "$VERSION"
```
Comment (zyga) [inline build-aux/snap/snapcraft.yaml:229]: Nitpick, for sanity reasons I'd strip -fips and just set the version without calling mkversion.sh again.

**Review by zyga — APPROVED** [review]
LGTM with one suggestion

File: build-aux/snap/snapcraft.yaml (hunk 1 above)
Comment (bboozzoo) [reply build-aux/snap/snapcraft.yaml:229]: pull step called mkversion with --output-only, so no version files were generated, so we need to call it again

**Review by MiguelPires — APPROVED** [review]
lgtm

**Comment by bboozzoo** [comment]
Added skip spread and re-running the CI

**Pull request merged**